### ISSUE TITLE
Implement retraining loop with different techniques

### ISSUE THREAD

**Issue by leaszcz** [Uni-Tubingen-Teamproject/Network-Pruning-and-Interpretability#56]
### Describe the solution you’d like
We want to implement a retraining loop for our pruned model, exploring different retraining techniques. 

### Additional context
-	Use the code for global unstructured L1 Pruning as a starting point
-	Try to enable auxiliary heads in the model for retraining
-	Use nn.CrossEntropyLoss() as the criterion
-	For the loss function, take the weighted sum of the main loss and 0.3 times the auxiliary loss
-	Keep the masking layers on during retraining
-	think about possible termination criterions
-	e.g. CLR Retraining
